csrc/ir/internal_nodes.h

```diff
@@ -2327,4 +2327,77 @@ class LinearOp : public Expr {
   }
 };
 
+// SDPA node with same functionality as F.scaled_dot_product_attention
+// https://pytorch.org/docs/stable/generated/torch.nn.functional.scaled_dot_product_attention.html
+class SdpaOp : public Expr {
+ public:
+  using Expr::Expr;
+
+  SdpaOp(
+      IrBuilderPasskey,
+      Val* out,
+      Val* query,
+      Val* key,
+      Val* value,
+      Val* attn_mask,
+      double dropout_p,
```

Comment:
Should we instead let `dropout_p` be a `Val*`?

Reply:
I agree with @jacobhinkle here.

Since we already allow things like `attn_mask` to be dynamic, we might as well also extend that to all arguments. Including `is_causal`.

IIUC, given current implementation, if we are going to want to schedule and codegen sdpa in the future, we'll have to concretize on attn_mask anyway, since the computational pattern would change based on whether or not we'll have an attn_mask. Why stop there and not extend this all the way to other arguments?